rllib/models/tf/modelv1_compat.py

```diff
@@ -124,18 +123,21 @@ def value_function(self):
                     # Create a new separate model with no RNN state, etc.
                     branch_model_config = self.model_config.copy()
                     branch_model_config["free_log_std"] = False
+                    input_dict = self.cur_instance.input_dict
                     if branch_model_config["use_lstm"]:
                         branch_model_config["use_lstm"] = False
-                        if log_once("vf_warn"):
-                            logger.warning(
-                                "It is not recommended to use a LSTM model "
-                                "with vf_share_layers=False (consider setting "
-                                "it to True). If you want to not share "
-                                "layers, you can implement a custom LSTM "
-                                "model that overrides the value_function() "
-                                "method.")
+                        logger.warning(
+                            "It is not recommended to use an LSTM model "
+                            "with the `vf_share_layers=False` option. "
+                            "If you do not want to use a shared policy+vf "
+                            "network, you can implement a custom LSTM "
+                            "model that overrides the value_function() "
+                            "method. "
+                            "`use_lstm` has been set to False for the "
+                            "vf-part of your network.")
+                        input_dict["obs"] = np.array([self.obs_space.sample()])
```

Comment:
Isn't this not a tensor?

Reply:
Yes, but this is only for the first dummy pass, correct?

Reply:
You are right. I was thinking tf2.x :/
This will screw up the static graph. I'll take another look.

Reply:
This is resolved now: vf-NN takes LSTM's last output as input (same as for non-Tuple observations). The problem was the preprocessor/shape check, which failed. This check only happened for Tuple/Dict observations, but is not necessary in the case of a shared vf (b/c no preprocessing of the LSTM output necessary).